wpimath/src/main/native/include/frc/estimator/PoseEstimator.h

```diff
@@ -136,24 +136,68 @@ class WPILIB_DLLEXPORT PoseEstimator {
    *
    * @param translation The pose to translation to.
    */
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif  // defined(__GNUC__) && !defined(__clang__)
   void ResetTranslation(const Translation2d& translation) {
     m_odometry.ResetTranslation(translation);
+
+    std::optional<std::pair<units::second_t, VisionUpdate>> latestVisionUpdate =
+        m_visionUpdates.empty() ? std::nullopt
+                                : std::optional{*m_visionUpdates.crbegin()};
     m_odometryPoseBuffer.Clear();
     m_visionUpdates.clear();
-    m_poseEstimate = m_odometry.GetPose();
+
+    if (latestVisionUpdate) {
+      // apply vision compensation to the pose rotation
+      VisionUpdate const visionUpdate{
```

Comment:
Isn't the normal convention `const VisionUpdate visionUpdate` instead of `VisionUpdate const visionUpdate`?